### PR TITLE
Feature: Add Vote replacement tests

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

Vote replacement was already supported by the  vote function so tests were added to test this functionmality

Task: [APP-1266](https://aragonassociation.atlassian.net/browse/APP-1266)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder of the package after the [UPCOMING] title and before the latest version.
- [ ] I have tested my code on the test network.

[APP-1266]: https://aragonassociation.atlassian.net/browse/APP-1266?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/client/test/integration/client-addressList/methods.test.ts [hunk 1]
```diff
@@ -134,11 +134,37 @@ describe("Client Address List", () => {
       const client = new ClientAddressList(ctxPlugin);
 
       const voteParams: IVoteProposalParams = {
-        pluginAddress: "0x1234567890123456789012345678901234567890",
-        proposalId: "0x1234567890123456789012345678901234567890",
+        pluginAddress,
+        proposalId: "0x0",
```
Comment (brickpop) [inline modules/client/test/integration/client-addressList/methods.test.ts]: how can it work with ID 0x0?

Comment (josemarinas) [reply modules/client/test/integration/client-addressList/methods.test.ts]: The contracts accept a bigNumberish, this means `string | number | bigint | hex string| bignumber` so literally atm they dont care once we implement the new proposalID with validation this wont work and will need to be updated

Comment (brickpop) [reply modules/client/test/integration/client-addressList/methods.test.ts]: Ok, but as of today how can this work? Is the current proposalId 0x0 valid and open for vote?

If so, resolve and merge. I'm good for the rest :+1:

**Review by brickpop — APPROVED** [review]

File: modules/client/test/integration/client-addressList/methods.test.ts (hunk 1 above)
Comment (josemarinas) [reply modules/client/test/integration/client-addressList/methods.test.ts]: Yes, its is created in the previous test

**Pull request merged**